torch/nn/init.py

```diff
@@ -525,3 +528,97 @@ def deprecated_init(*args, **kwargs):
 kaiming_normal = _make_deprecate(kaiming_normal_)
 orthogonal = _make_deprecate(orthogonal_)
 sparse = _make_deprecate(sparse_)
+
+
+GET_SEMANTIC_VERSION = re.compile(r'\d+\.\d+\.\d+')
+
+def _get_semantic_version_from_string(version: str) -> Tuple[int, int, int]:
+    """version: should be 'major.minor.micro'"""
+    v = GET_SEMANTIC_VERSION.findall(version)
+
+    if len(v) != 1:
+        raise TypeError("Invalid version, not a valid semantic version (e.g. of correct input 1.7.1)")
+
+    v = v[0].split('.')
+    return (int(v[0]), int(v[1]), int(v[2]))
+
+
+# Use "init_version" to handle "_init_version". This value corresponds to the last pytorch version that
+# did not have 'init_version' context manager.
+_init_version = (1, 6, 1)
+_torch_version = _get_semantic_version_from_string(torch.__version__)
+
+def _parse_version(version: Union[Tuple[int, int, int], str] = None) -> Tuple[int, int, int]:
+    if version is None:
+        return _init_version
+
+    if isinstance(version, tuple):
+        for x in version:
+            if not isinstance(x, int):
+                raise ValueError("Invalid version, must be a version string (e.g. '1.7.0') or tuple of integers")
+    elif isinstance(version, str):
+        version = _get_semantic_version_from_string(version)
+    else:
+        raise TypeError("Invalid version, must be a version string (e.g. '1.7.0') or tuple of integers")
+
+    # version, should be less than torch.__version__
+    if version > _torch_version:
+        raise ValueError(f"version {version} should be less than torch version {_torch_version}")
+    return version
+
+
+@contextlib.contextmanager
+def init_version(version=None):
+    r"""Context manager to use a specific version of initialization for `nn.modules`.
+    By default, the pytorch initialization used till version 1.6.1 is used.
+
+    Args:
+        version : Union[Tuple[int, int, int], str], which pytorch version to use for initializing nn.modules.
+            The format should be a version string (major,minor,micro) or a tuple of integers. Examples of valid
+            version are (1,7,0), '1.7.1'.
+
+    Examples:
+        >>> with nn.init.init_version() as version:
+        >>>     # use the default pytorch initialization used till pytorch version 1.6.1
+        >>>     # This is the default behaviour.
+        >>>     model = nn.Sequential(...)
```

Comment:
Removed this example. I think it is a redundant example. Also, it might create confusion if we were to change the functionality. \
I think the best use case of context manger is to provide a version to it or set the global.